libraries/Microsoft.Bot.Builder.Dialogs/Prompts/OAuthPrompt.cs

```diff
@@ -0,0 +1,95 @@
+﻿using System;
+using prompts = Microsoft.Bot.Builder.Prompts;
+using static Microsoft.Bot.Builder.Prompts.PromptValidatorEx;
+using Microsoft.Bot.Schema;
+using System.Threading.Tasks;
+using Microsoft.Bot.Builder.Prompts;
+
+namespace Microsoft.Bot.Builder.Dialogs
+{
+    public class OAuthPromptSettingsWithTimeout : prompts.OAuthPromptSettings
+    {
+        public int? Timeout { get; set; }
+    }
+
+    public class OAuthPromptState : PromptOptions
+    {
+        public OAuthPromptState(PromptOptions defaultPromptOptions) : base()
+        {
+            if (defaultPromptOptions != null)
+            {
+                PromptString = PromptString ?? defaultPromptOptions.PromptString;
+                PromptActivity = PromptActivity ?? defaultPromptOptions.PromptActivity;
+                Speak = Speak ?? defaultPromptOptions.Speak;
+                RetryPromptString = RetryPromptString ?? defaultPromptOptions.RetryPromptString;
+                RetryPromptActivity = RetryPromptActivity ?? defaultPromptOptions.RetryPromptActivity;
+                RetrySpeak = RetrySpeak ?? defaultPromptOptions.RetrySpeak;
+            }
+        }
+
+        public DateTime Expires { get; set; }
+    }
+
+    public class OAuthPrompt : Control, IDialog, IDialogContinue
+    {
+        private prompts.OAuthPrompt _prompt;
+        private OAuthPromptSettingsWithTimeout _settings;
+
+        public OAuthPrompt(OAuthPromptSettingsWithTimeout settings, PromptValidator<TokenResult> validator = null)
+        {
+            _settings = settings ?? throw new ArgumentException(nameof(settings));
+            _prompt = new prompts.OAuthPrompt(settings, validator);
+        }
+
+        public async Task DialogBegin(DialogContext dc, object dialogArgs = null)
+        {
+            if (dc == null)
+                throw new ArgumentNullException(nameof(dc));
+            if (dialogArgs == null)
```

Comment:
This doesn't make sense. you're *allowing* a default argument of null to be passed in:
`object dialogArgs = null`
... and then throwing an Argument Null Exception if they do. That's wrong. 

This tell me there is no Unit tests around this. Those are needed.

Reply:
agreed. however, i am just planning on getting this port of the node code in first and then start adding tests and samples.

---
In reply to: [184818595](https://github.com/Microsoft/botbuilder-dotnet/pull/520#discussion_r184818595) [](ancestors = 184818595)